Bukkit/src/main/java/com/plotsquared/bukkit/listener/BlockEventListener.java

```diff
@@ -324,6 +326,30 @@ public void blockCreate(BlockPlaceEvent event) {
                     return;
                 }
             }
+            // The other block part can be placed outside the plot (#3483)
+            if (event.getBlock().getBlockData() instanceof Bed bed) {
+                org.bukkit.Location other = event.getBlock().getLocation().add(bed.getFacing().getDirection());
+                if (!(Objects.equals(BukkitUtil.adapt(other).getPlot(), plot))) {
+                    pp.sendMessage(
+                            TranslatableCaption.of("permission.no_permission_event"),
+                            Template.of("node", String.valueOf(Permission.PERMISSION_ADMIN_BUILD_ROAD))
+                    );
+                    event.setCancelled(true);
+                    return;
+                }
+            }
+            // Same goes for doors, which may exceed the area height limit
```

Comment:
Good catch with the doors! Though I can imagine that the same applies to more items? E.g. horizontally placed item frames or trees grown from bonemeal interaction?

Reply:
I couldn't think about more items witch multiple block states - If you have a list of all blocks and interactions I could try to implement those

Reply:
I created a new issue to not clutter this PR, considering it's somewhat out of the scope of this PR https://github.com/IntellectualSites/PlotSquared/issues/3488
Would be cool if you would want to work on it 👍🏻